src/clusterfuzz/_internal/bot/tasks/utasks/uworker_errors.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module for enumerating errors in utasks."""
+import enum
+
+
+class Type(enum.Enum):
+  """Errors during utask_main."""
+  ANALYZE_BUILD_SETUP = 1
+  ANALYZE_NO_CRASH = 2
+  TESTCASE_SETUP = 3
+  NO_FUZZER = 4
+
+
+class Error:
+  """Class representing error messages from the untrusted worker. This should
+  contain the type of the error |error_type| as well as any other data to handle
+  the error."""
+
+  def __init__(self, error_type, **kwargs):
+    self.error_type = error_type
+    assert error_type not in kwargs
```

Comment:
Just out of curiosity: Would python ever allow users to have another `error_type` in `kwargs` given there is a parameter under the same name?

Reply:
Also, instead of having an assertion failure when `error_type` is in `kwargs`, how about popping out the `error_type` and log a waning/erorr?
I noticed that later code pops it before invoking this function, and I guess an alternative is to always do that in one place here, instead of remembering to do that every time before constructing an `Error`?

Reply:
Youre right that Python doesn't allow this.
I think assert failing makes much more sense than warning here. This is a programmer error that should be fixed and it's unclear how to even have it. Very different from something like a network error.